sdk/eventhub/azure-eventhub/stress/azure_eventhub_producer_stress.py

```diff
@@ -24,8 +25,13 @@ def stress_send_sync(producer: EventHubProducerClient, args, logger):
             event_data = EventData(body=b"D" * args.payload)
             batch.add(event_data)
     except ValueError:
-        producer.send_batch(batch)
-        return len(batch)
+        try:
+            producer.send_batch(batch)
+        except EventHubError as e:
+            logger.warning("Sync send failed due to error: %r. Retrying...", e)
```

Comment:
If this happens persistently, do we have any way to alert?  (e.g. if internet cut/service outage)  Past a point it would seem to invalidate the test; while this is a somewhat pathological scenario ("I'm being paranoid") it's also something I'd be nervous to not be aware of.

Reply:
yeah, I totally agree with what you said.

So the case is that: when we run 3-days stress test, chances are that error happens on service side (like you mentioned outage/network/etc.).
It's not sdk's fault but it leads to the stress test stopped.

The original goal of this improvement was to avoid this kind of situation, and we check logs after the stress test to see if everything is good.

But after reading your concern, I think this part can be further improved, like having two modes of sending:

- do_normal_send_and_stop_if_there_is_an_error
- just_keep_sending_I_will_check_log_by_myself

does this make senses to you?

Reply:
the parameter is optional now which at least provides flexibility to some extent.